packages/split-button/src/Menu/Menu.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { MouseEvent, MouseEventHandler, useRef, useState } from 'react';
+import isEmpty from 'lodash/isEmpty';
+
+import Button from '@leafygreen-ui/button';
+import { cx } from '@leafygreen-ui/emotion';
+import { useBackdropClick, useEventListener } from '@leafygreen-ui/hooks';
+import Icon from '@leafygreen-ui/icon';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+import { isComponentType, keyMap } from '@leafygreen-ui/lib';
+import { Menu as LGMenu } from '@leafygreen-ui/menu';
+
+import { MenuItemType } from '../SplitButton';
+
+import {
+  triggerBaseStyles,
+  triggerSizeStyles,
+  triggerThemeStyles,
+} from './Menu.styles';
+import { MenuProps } from './Menu.types';
+
+/**
+ * @internal
+ */
+export const Menu = ({
+  variant,
+  disabled,
+  align,
+  justify,
+  size,
+  baseFontSize,
+  menuItems,
+  containerRef,
+  id,
+  onTriggerClick,
+  triggerAriaLabel,
+  open: controlledOpen,
+  setOpen: controlledSetOpen,
+  ...rest
+}: MenuProps) => {
+  const { theme } = useDarkMode();
+  const [uncontrolledOpen, uncontrolledSetOpen] = useState<boolean>(false);
+  const buttonRef = useRef<null | HTMLButtonElement>(null);
+  const menuRef = useRef<null | HTMLDivElement>(null);
+
+  const setOpen =
```

Comment:
does the `useControlledValue` hook work here?

Reply:
no, that's for input values. We could make a new hook for open/setOpen.